File: NetBeans_IDE/MmgGameApiJava/cfg/asset_src/dungeon_trap_level1_classes/MdtItemPotion.java
package game.jam.DungeonTrap;

import net.middlemind.MmgGameApiJava.MmgBase.MmgBmp;

/**
 * A class that represents a yellow potion.
 * 
 * @author Victor G. Brusca, Middlemind Games
 * 09/19/2020
 */
public class MdtItemPotion extends MdtItem {
    
    /**
     * The type of potion this class represents.
     */
    public MdtItemPotionType potionType = MdtItemPotionType.NONE;
        
    /**
     * A basic constructor for the yellow potion class.
     * 
     * @param Subj          The subject to use for this potion.
     * @param PotionType    The potion type to use for this item.
     * @param Points        The points type to use for this potion.
     */
    public MdtItemPotion(MmgBmp Subj, MdtItemPotionType PotionType, MdtPointsType Points) {
        super(Subj, MdtObjType.ITEM, MdtObjSubType.ITEM_POTION, Points);
        SetPotionType(PotionType);
    }

    /**
     * Gets the subject of the class.
     * 
     * @return      The subject of the class. 
     */
    @Override
    public MmgBmp GetSubj() {
        return (MmgBmp)subj;
    }

    /**
     * Gets the potion type of this object.
     * 
     * @return      The potion type of this object.
     */
    public MdtItemPotionType GetPotionType() {
        return potionType;
    }
    
    /**
     * Sets the potion type of this object.
     * 
     * @param PotionType    The potion type of this object. 
     */
    public void SetPotionType(MdtItemPotionType PotionType) {
        potionType = PotionType;
    }
}

File: VisualStudio_IDE/MmgGameApiCs/MmgGameApiCs/src/net/middlemind/DungeonTrap/ChapterE5_Phase2/MdtItemPotion.cs
using System;
using net.middlemind.MmgGameApiCs.MmgBase;

namespace net.middlemind.DungeonTrap.ChapterE5_Phase2
{
    /// <summary>
    /// A class that represents a yellow potion.
    ///
    /// @author Victor G.Brusca, Middlemind Games
    /// 09/19/2020
    /// </summary>
    public class MdtItemPotion : MdtItem
    {
        /// <summary>
        /// The type of potion this class represents.
        /// </summary>
        public MdtItemPotionType potionType = MdtItemPotionType.NONE;

        /// <summary>
        /// A basic constructor for the yellow potion class.
        /// </summary>
        /// <param name="Subj">The subject to use for this potion.</param>
        /// <param name="PotionType">The potion type to use for this item.</param>
        /// <param name="Points">The points type to use for this potion.</param>
        public MdtItemPotion(MmgBmp Subj, MdtItemPotionType PotionType, MdtPointsType Points)
            : base(Subj, MdtObjType.ITEM, MdtObjSubType.ITEM_POTION, Points)
        {
            SetPotionType(PotionType);
        }

        /// <summary>
        /// Gets the subject of the class.
        /// </summary>
        /// <returns>The subject of the class.</returns>
        public new MmgBmp GetSubj()
        {
            return (MmgBmp)subj;
        }

        /// <summary>
        /// Gets the potion type of this object.
        /// </summary>
        /// <returns>The potion type of this object.</returns>
        public virtual MdtItemPotionType GetPotionType()
        {
            return potionType;
        }

        /// <summary>
        /// Sets the potion type of this object.
        /// </summary>
        /// <param name="PotionType">The potion type of this object.</param>
        public void SetPotionType(MdtItemPotionType PotionType)
        {
            potionType = PotionType;
        }
    }
}